### PR TITLE
Update hyper to 0.14.4 (uplift to 1.20.x)

### PR DESCRIPTION
**Pull request by antonok-edm**
Manual uplift of #7853, also rolling up changes from #7740 and #7727
Resolves brave/brave-browser#14009

Approved, please ensure that before merging:

- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR.
- [ ] You have tested your change on Nightly.
- [x] The PR milestones match the branch they are landing to.

After you merge:

- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by antonok-edm** [comment]
apologies for reviewer request noise, I opened this against `master` initially :sweat:

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.20.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
